doc/source/contributing_code.rst

```diff
@@ -212,18 +214,37 @@ First we need to install dependencies. You'll need a mix of conda and pip instal
 
     .. code-block:: bash
 
-        conda install -c bioconda -c conda-forge \
-            biopython chemfiles clustalw==2.1 codecov cython \
-            griddataformats gsd hypothesis "joblib>=0.12" \
-            matplotlib mmtf-python mock netcdf4 networkx \
-            "numpy>=1.18.0" psutil pytest scikit-learn scipy \
-            "seaborn>=0.7.0" sphinx "tidynamics>=1.0.0" \
-            "tqdm>=4.43.0"
+        conda install -c conda-forge \
```

Comment:
I think it would be more convenient to use an environment.yaml file from the repo. Then we only need to keep _that_ file up to date.

Reply:
I think there was a discussion about it on core at some point. The main thing was that we had a ton of different files lying around that didn't make sense (for example the RTD yaml file is pinned to different version numbers because of reasons I've not had a time to dig into yet).

What we could probably do is have a standard user facing one and have CI validate it (by picking up the setup-deps action in some way or another).

Reply:
Along the lines of "check if developer.yaml is compatible with the env that we are building via setup-deps"? Sounds like a good CI check.

Perhaps just build the env with setup-deps, then try to run mamba with the yaml file and check that _nothing_ actually gets updated by diffing the env list before and after.

Reply:
yeah something like that should work